main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala

```diff
@@ -154,7 +175,14 @@ object EffUnification3 {
       m.getForward(atom) match {
         case None => throw InternalCompilerException(s"Unexpected unbound type variable: '$tpe'.", tpe.loc)
         case Some(x) => atom match {
-          case Atom.VarFlex(_) => SetFormula.Var(x) // A flexible variable is a real variable.
+          case Atom.VarFlex(sym) =>
+            if (sym.isSlack && !withSlack) {
```

Comment:
I guess when we go back, these syms won't appear in the substitution. But should they instead be mapped to empty?